lib/modules/onc_program/onc_standalone_public_launch_sequence.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+require_relative './shared_onc_launch_tests'
+
+module Inferno
+  module Sequence
+    class OncStandalonePublicLaunchSequence < SequenceBase
+      include Inferno::Sequence::SharedONCLaunchTests
+
+      title 'Public Client Standalone Launch Sequence'
+
+      description 'Register Inferno as a public client with patient access and execute standalone launch.'
+
+      test_id_prefix 'OSLSP'
+
+      requires :onc_public_client_id,
+               :onc_public_scopes,
+               :onc_sl_oauth_authorize_endpoint,
+               :onc_sl_oauth_token_endpoint,
+               :initiate_login_uri,
+               :redirect_uris
+
+      defines :token, :id_token, :refresh_token, :patient_id
+
+      show_uris
+
+      def valid_resource_types
+        [
+          '*',
+          'Patient',
+          'AllergyIntolerance',
+          'CarePlan',
+          'CareTeam',
+          'Condition',
+          'Device',
+          'DiagnosticReport',
+          'DocumentReference',
+          'Encounter',
+          'Goal',
+          'Immunization',
+          'Location',
+          'Medication',
+          'MedicationOrder',
+          'MedicationRequest',
+          'MedicationStatement',
+          'Observation',
+          'Organization',
+          'Practitioner',
+          'PractitionerRole',
+          'Procedure',
+          'Provenance'
+        ]
+      end
+
+      def required_scopes
+        ['launch/patient']
+      end
+
+      details %(
+        # Background
+
+        The [Standalone
+        Launch](http://hl7.org/fhir/smart-app-launch/#standalone-launch-sequence)
+        Sequence allows an app, like Inferno, to be launched independent of an
+        existing EHR session. It is one of the two launch methods described in
+        the SMART App Launch Framework alongside EHR Launch. The app will
+        request authorization for the provided scope from the authorization
+        endpoint, ultimately receiving an authorization token which can be used
+        to gain access to resources on the FHIR server.
+
+        # Test Methodology
+
+        Inferno will redirect the user to the the authorization endpoint so that
+        they may provide any required credentials and authorize the application.
+        Upon successful authorization, Inferno will exchange the authorization
+        code provided for an access token.
+
+        For more information on the #{title}:
+
+        * [Standalone Launch Sequence](http://hl7.org/fhir/smart-app-launch/#standalone-launch-sequence)
+      )
+
+      def url_property
+        'onc_sl_url'
+      end
+
+      def instance_url
+        @instance.send(url_property)
+      end
+
+      def instance_client_id
+        @instance.onc_public_client_id
+      end
+
+      def instance_confidential_client
+        false
+      end
+
+      def instance_client_secret
```

Comment:
I don't think this method is necessary and it seems weird to declare this in a public launch file (I understand we may need to for some shared launch tests).  I looked in the shared_onc_launch tests and all the references to this were wrapped in a check for instance_confidential_client so I think it's safe to delete.

Reply:
I did it just to be thorough and to be clear that this test won't be using a client_secret.  It defaults to the client secret used by the EHR launch, and it seems risky to just assume that we have enough guards to not send that along. I think this is ok to leave.

As is probably pretty clear, we have a good opportunity here to make what I"m trying to do much more straightforward than it is.  So at a certain point, I'm just accepting that this is hard to read, but functional, and we can consider what this is doing for future designs.

Reply:
I don't agree that it makes it clear that it won't be using a client_secret (if anything it makes me suspicious that it is!), but I understand the concern about guards.  Although, if those guards are not properly instituted this might hide other issues (like an `Authorization` header being set).

Seeing as we don't have much time to fix this I'm happy with making this more straightforward and logical at a later date.